fem/quadinterpolator.hpp

```diff
@@ -99,26 +105,36 @@ class QuadratureInterpolator
              Vector &q_val, Vector &q_der, Vector &q_det) const;
 
    /// Interpolate the values of the E-vector @a e_vec at quadrature points.
+   template <QVectorLayout>
+   void Values(const Vector &e_vec, Vector &q_val) const;
    void Values(const Vector &e_vec, Vector &q_val) const;
 
    /** @brief Interpolate the derivatives of the E-vector @a e_vec at quadrature
        points. */
+   template <QVectorLayout>
+   void Derivatives(const Vector &e_vec, Vector &q_der) const;
    void Derivatives(const Vector &e_vec, Vector &q_der) const;
 
    /** @brief Interpolate the derivatives in physical space of the E-vector
        @a e_vec at quadrature points. */
+   template <QVectorLayout>
    void PhysDerivatives(const Vector &e_vec, Vector &q_der) const;
+   void PhysDerivatives(const Vector &e_vec, Vector &q_der) const;
+
+   /// Compute the determinant of the E-vector @a e_vec at quadrature points.
+   void Determinants(const Vector &e_vec, Vector &q_det) const;
 
    /// Perform the transpose operation of Mult(). (TODO)
    void MultTranspose(unsigned eval_flags, const Vector &q_val,
                       const Vector &q_der, Vector &e_vec) const;
 
    // Compute kernels follow (cannot be private or protected with nvcc)
-
    /// Template compute kernel for 2D.
    template<const int T_VDIM = 0, const int T_ND = 0, const int T_NQ = 0>
-   static void Eval2D(const int NE,
+   static void Mult2D(const int NE,
                       const int vdim,
+                      const QVectorLayout q_layout,
+                      const GeometricFactors *geom,
```

Comment:
It is probably better to keep the old method, marked as deprecated. I assume it can be implemented by just calling the new method?

Reply:
Keep the old method.